### PR TITLE
feat: add comulative data for volume and trades

### PR DESCRIPTION
**Pull request by guerrap**
# Summary

Relative to #41 

Done:
- Bit of layout change to the dashboard page
- Load each component in the dashboard separately
- Add cumulative `Volume` data card
- Add cumulative `Trades` data card

image

image

# How To Test
1.  Open the page `Dashboard`
- [ ] Verify that `All time volume` and `Transactions` data card exist.

# Background 

There are also changes that include the transactions historical data, that are not required for the cumulative data cards but are needed for the new transactions card (WIP)

### CODE REVIEW

File: src/contexts/Dashboard.js [hunk 1]
```diff
@@ -117,9 +153,136 @@ export function useDashboardChartData() {
...
+
+/**
+ * Get historical data for transactions for each network (last 1 month)
+ */
+const getTransactions = async () => {
+  try {
+    let transactions = [];
+
+    const utcCurrentTime = dayjs();
+    const utcThreeMonthsBack = utcCurrentTime.subtract(1, 'month').startOf('minute').unix();
```
Comment (wixzi) [inline src/contexts/Dashboard.js:202]:
```suggestion
    const utcOneMonthsBack = utcCurrentTime.subtract(1, 'month').startOf('minute').unix();
```

File: src/pages/DashboardPage.js [hunk 2]
```diff
@@ -1,37 +1,60 @@
...
+
+const PanelLoaderWrapper = ({ isLoading, children }) => <Panel>{isLoading ? <LocalLoader /> : children}</Panel>;
+
 const DashboardPage = () => {
   const chartData = useDashboardChartData();
+  const comulativeData = useDashboardComulativeData();
   const [formattedLiquidityData, setFormattedLiquidityData] = useState([]);
   const [formattedVolumeData, setFormattedVolumeData] = useState([]);
+  const [formattedTotalTrades, setFormattedTotalTrades] = useState([]);
+  const [formattedTotalVolume, setFormattedTotalVolume] = useState([]);
```
Comment (wixzi) [inline src/pages/DashboardPage.js:49]: use a single state and use two keys in place of `value` like `trades` and `volumes`

File: src/pages/DashboardPage.js [hunk 3]
```diff
@@ -52,54 +75,141 @@ const DashboardPage = () => {
...
+        .filter((key) => Object.values(SupportedNetwork).includes(key))
+        .forEach((network) => {
+          totalTradesPerNetwork.push({
+            network,
+            value: formattedNum(comulativeData[network].totalTrades),
+          });
+          totalVolumePerNetwork.push({
+            network,
+            value: `$ ${formattedNum(comulativeData[network].totalVolume)}`,
+          });
```
Comment (wixzi) [inline src/pages/DashboardPage.js:99]:
```suggestion
          totalTradeAndVolumesPerNetwork.push({
            network,
             volume: formattedNum(comulativeData[network].totalVolume),
            trade: formattedNum(comulativeData[network].totalTrades),
          });
       
```

File: src/pages/DashboardPage.js [hunk 4]
```diff
@@ -52,54 +75,141 @@ const DashboardPage = () => {
...
+                      networksValues={formattedTotalVolume}
+                    />
+                  </PanelLoaderWrapper>
+                  <PanelLoaderWrapper isLoading={isTotalTradesLoading}>
+                    <ComulativeNetworkDataCard
+                      title={'Total trades'}
+                      icon={<FileText size={22} color={'#50dfb6'} />}
+                      comulativeValue={formattedNum(comulativeData.totalTrades)}
+                      networksValues={formattedTotalTrades}
+                    />
```
Comment (wixzi) [inline src/pages/DashboardPage.js:173]: then you can pass the key.

File: src/contexts/Dashboard.js [hunk 5]
```diff
@@ -48,6 +47,24 @@ function reducer(state, { type, payload }) {
       };
     }
 
+    case UPDATE_COMULATIVE_DATA: {
+      const { comulativeNetworksData } = payload;
+      return {
+        ...state,
+        comulativeData: {
+          ...comulativeNetworksData,
+        },
```
Comment (wixzi) [inline src/contexts/Dashboard.js:56]:
```suggestion
        comulativeData: comulativeNetworksData,
```

File: src/pages/DashboardPage.js (hunk 3 above)
Comment (guerrap) [reply src/pages/DashboardPage.js:99]: I think it's better to accumulate them in 2 different arrays and then set them in a single state as you suggested, something like:
```js
const totalTradesPerNetwork = [];
const totalVolumePerNetwork = [];

Object.keys(comulativeData)
  // include only network specific data
  .filter((key) => Object.values(SupportedNetwork).includes(key))
  .forEach((network) => {
    totalTradesPerNetwork.push({
      network,
      value: formattedNum(comulativeData[network].totalTrades),
    });
    totalVolumePerNetwork.push({
      network,
      value: `$ ${formattedNum(comulativeData[network].totalVolume)}`,
    });
  });

setFormattedComulativeData({ trades: totalTradesPerNetwork, volume: totalVolumePerNetwork })
```
So in the components it's simpler to pass the required values:
```jsx
<ComulativeNetworkDataCard
  title={'Total trades'}
  icon={<FileText size={22} color={'#50dfb6'} />}
  comulativeValue={formattedNum(comulativeData.totalTrades)}
  networksValues={formattedComulativeData.trades}
/>
```
```jsx
<ComulativeNetworkDataCard
  title={'All time volume'}
  icon={<DollarSign size={22} color={'#50dfb6'} />}
  comulativeValue={`$ ${formattedNum(comulativeData.totalVolume)}`}
  networksValues={formattedComulativeData.volume}
/>
```

**Review by wixzi — APPROVED** [review]

**Pull request merged**